Packages/com.chisel.editor/Chisel/Editor/Components/Base/Editor/ChiselNodeEditor.cs

```diff
@@ -224,50 +204,139 @@ public Styles()
 
         static bool Toggle(bool selected, GUIContent[] content, GUIStyle[] style)
         {
-            return GUILayout.Button(selected ? content[1] : content[0], selected ? style[1] : style[0]);
+            var selectedContent = selected ? content[1] : content[0];
+            var selectedStyle = selected ? style[1] : style[0];
+
+            return GUILayout.Button(selectedContent, selectedStyle);
+        }
+
+        public static void ShowOperationChoices(SerializedProperty operationProp)
+        {
+            EditorGUILayout.BeginHorizontal();
+            ShowOperationChoicesInternal(operationProp);
+            EditorGUILayout.EndHorizontal();
         }
 
         // TODO: put somewhere else
-        internal static void ShowOperationChoices(SerializedProperty operationProp)
+        public static void ShowOperationChoicesInternal(SerializedProperty operationProp)
         {
             if (styles == null)
                 styles = new Styles();
-            EditorGUILayout.BeginHorizontal();
-            var operation = operationProp.hasMultipleDifferentValues ? ((CSGOperationType)255) : ((CSGOperationType)operationProp.enumValueIndex);
-            if (Toggle((operation == CSGOperationType.Additive)     , styles.AdditiveIcon, styles.leftButton))
-                operationProp.enumValueIndex = (int)CSGOperationType.Additive;
-            if (Toggle((operation == CSGOperationType.Subtractive)	, styles.SubtractiveIcon , styles.midButton))
-                operationProp.enumValueIndex = (int)CSGOperationType.Subtractive;
-            if (Toggle((operation == CSGOperationType.Intersecting) , styles.IntersectingIcon, styles.rightButton))
-                operationProp.enumValueIndex = (int)CSGOperationType.Intersecting;
-            //EditorGUILayout.PropertyField(operationProp);
-            EditorGUILayout.EndHorizontal();
+
+            const string AdditiveIconName = "csg_addition";
+            const string SubtractiveIconName = "csg_subtraction";
+            const string IntersectingIconName = "csg_intersection";
+
+            const string AdditiveIconTooltip = "Additive CSG Operation";
+            const string SubtractiveIconTooltip = "Subtractive CSG Operation";
+            const string IntersectingIconTooltip = "Intersecting CSG Operation";
+
+            var additiveIcon = ChiselEditorResources.GetIconContent(AdditiveIconName, AdditiveIconTooltip);
+            var subtractiveIcon = ChiselEditorResources.GetIconContent(SubtractiveIconName, SubtractiveIconTooltip);
+            var intersectingIcon = ChiselEditorResources.GetIconContent(IntersectingIconName, IntersectingIconTooltip);
+
+            using (new EditorGUIUtility.IconSizeScope(new Vector2(16, 16)))     // This ensures that the icons will be the same size on regular displays and HDPI displays
```

Comment:
When in Retina mode it'll load higher resolution images for the icons.
Unfortunately Unity hasn't exposed functionality to user code that automatically resizes the icons for us in labels, so any larger image in user land would be rendered larger instead of at a higher resolution.
To fix this we use IconSizeScope to make sure the icons are rendered at the correct resolution.